From ed01d2d5236d7f94b6566efa51c1223402a6f3a1 Mon Sep 17 00:00:00 2001
From: Igal Zeltser <izeltser@habana.ai>
Date: Tue, 08 Jun 2021 17:27:23 +0300
Subject: [PATCH] [SW-44814] pci: reading mem enable status from aux block

So far the memory access enable status was read (polled) from the
PCIE_DBI block register. Testing shows that if this register is
being read during pci link transition it can cause a cpu freeze.
To avoid this, the memory access enable status is being polled
from the PCIE_AUX block register PCIE_AUX_MEM_SPACE_EN.

Change-Id: I2801a0b8c555519119aa128184c6c9097133e952
Signed-off-by: Igal Zeltser <izeltser@habana.ai>
---

diff --git a/common/pci.c b/common/pci.c
index c94654d..d4ef70a 100644
--- a/common/pci.c
+++ b/common/pci.c
@@ -174,21 +174,12 @@
 	return 0;
 }
 
-#define DBI_MEM_SPACE_EN_MASK	0x2
-
 /*
  * hl_pci_is_mem_enabled() - is memory access enabled in the configuration
  */
 __weak bool hl_pci_is_mem_enabled(void)
 {
-	u32 sts_cmd_reg;
-	u32 aux_dbi_reg = regmap_readl(mmPCIE_AUX_DBI);
+	u32 mem_space_ena_reg = regmap_readl(mmPCIE_AUX_MEM_SPACE_EN);
 
-	regmap_writel(IATU_ACCESS_DISABLE_VAL, mmPCIE_AUX_DBI);
-
-	sts_cmd_reg = regmap_readl(mmPCIE_DBI_STATUS_COMMAND_REG);
-
-	regmap_writel(aux_dbi_reg, mmPCIE_AUX_DBI);
-
-	return !!(sts_cmd_reg & DBI_MEM_SPACE_EN_MASK);
+	return !!mem_space_ena_reg;
 }
